syncstorage/src/db/mod.rs

```diff
@@ -69,13 +77,24 @@ pub fn spawn_pool_periodic_reporter<T: GetPoolState + Send + 'static>(
     Ok(())
 }
 
-pub async fn run_on_blocking_threadpool<F, T>(f: F) -> Result<T, DbError>
+/// Runs a function as a task on Actix's blocking threadpool.
+///
+/// WARNING: Calling `web::block` anywhere else will result in inaccurate threadpool metrics being
+/// reported. If you want to spawn a task on Actix's blocking threadpool, you **must** use this
+/// function.
+pub async fn run_on_blocking_threadpool<F, T>(metrics: Metrics, f: F) -> Result<T, DbError>
 where
     F: FnOnce() -> Result<T, DbError> + Send + 'static,
     T: Send + 'static,
 {
-    web::block(f).await.map_err(|e| match e {
+    metrics.incr_with_tag("blocking_tasks", "hostname", &HOSTNAME);
```

Comment:
This should probably be a `gauge` instead of a `counter`.

With an actual `gauge` we'd have to maintain the count on our side, however. An easy way would to do so would be to `Ordering::Relaxed` inc/dec say an `AtomicI64` somewhere (as much as I discourage globals, it could even be a module level static for this use case), and emit its value in our existing periodic reporter (which emits every 10 seconds, we could make that more often if necessary).

However I'll admit it's not totally clear to me how such different types ultimately result in influxdb/grafana (especially because the influxdb wire protocol treats everything as a number, it's just a matter of who's doing the tallying). Maybe the counter isn't much of a difference here. A `gauge` is at least be a bit more efficient for telegraf/influxdb to handle though.

Reply:
@pjenvey I hadn't heard of `gauge` before, but I agree that I like it here more than a counter 👍🏻 Do you think we should report changes to the gauge with every new blocking task, or should we spawn something that periodically reports the threadpool stats?

I'm thinking I could create a `BlockingThreadpool` struct that we keep in `ServerState` that has its own copy of `Metrics` and stores up-to-date counts of the number of blocking tasks. That would also get rid of the need for any globals

Reply:
@ethowitz I think the existing `db::spawn_pool_periodic_reporter` could emit it (but then maybe it needs to be renamed). It's currently hardcoded to report every 10 seconds. We could make that configurable if it needs to emit more often.